README.md

```diff
@@ -1418,6 +1401,40 @@ domain with a *.* like this instead:
 set :sessions, :domain => '.foo.com'
 ```
 
+#### Choosing Your Own Session Middleware
+
+Note that `enable :sessions` actually stores all data in a cookie. This
+might not always be what you want (storing lots of data will increase your
+traffic, for instance). You can use any Rack session middleware: in order to
+do so, one of the following methods can be used:
+
+```ruby
+enable :sessions
+set :session_store, Rack::Session::Pool
+```
+
+Or to enable sessions with a hash of options:
+
+```ruby
+set :sessions, :expire_after => 2592000
+set :session_store, Rack::Session::Pool
+```
+
+Another option is to **not** call `enable :sessions`, but instead pull in your
+middleware of choice as you would any other middleware:
+
+```ruby
+use Rack::Session::Pool, :expire_after => 2592000
+```
+
+It is important to note that when using this method, session based protection (see 'Configuring attack protection') will not be enabled by default. The Rack middleware to do that will also need to be added:
```

Comment:
I'm not sure I would mention `setup_protection`. It could be called but to avoid duplicate middleware being added it would have to set up like this:

```
disable :protection
use Rack::Session::Pool
setup_protection(self)
````
That would be the same as:

```
disable :protection
use Rack::Session::Pool
use Rack::Protection
```
Would you like me to also mention one of those methods?